klog_test.go

```diff
@@ -1493,6 +1597,63 @@ func (l *testLogr) WithValues(...interface{}) logr.Logger {
 	panic("not implemented")
 }
 
+type callDepthTestLogr struct {
+	testLogr
+	callDepth int
+}
+
+func (l *callDepthTestLogr) resetCallDepth() {
+	l.mutex.Lock()
+	defer l.mutex.Unlock()
+	l.callDepth = -1
```

Comment:
Why `-1`? Shouldn't that be `0` for "log the direct caller"?

Reply:
i think i had this to make sure i was doing things correctly when debugging tests, was using -1 as a sentinel, i'll put it back to 0